gcloud/_helpers.py

```diff
@@ -302,14 +303,30 @@ def _total_seconds(offset):
 def _rfc3339_to_datetime(dt_str):
     """Convert a string to a native timestamp.
 
+    .. note::
+
+       Python datetimes do not support nanosecond precision;  this function
+       therefore truncates such values to microseconds.
+
     :type dt_str: str
     :param dt_str: The string to convert.
 
     :rtype: :class:`datetime.datetime`
     :returns: The datetime object created from the string.
     """
-    return datetime.datetime.strptime(
-        dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    try:
+        return datetime.datetime.strptime(
+            dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    except ValueError:  # maybe nanosecond resoultion
+        prefix, suffix = dt_str.split('.')
+        fraction, zulu = suffix[:-1], suffix[-1]
```

Comment:
We'll end up with `IndexError` or similar out of this method, which was previously the stacktrace point for the `ValueError` being raised by `strptime`.  I'd say trying to parse any other format is YAGNI:  the logging API is the only one which was provoking errors, and its timestamps are uniformly nanosecond precision (as the [docs](https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/LogEntry#FIELDS.timestamp) specify).